src/all/namicomi/src/eu/kanade/tachiyomi/extension/all/namicomi/dto/MangaDto.kt

```diff
@@ -0,0 +1,101 @@
+package eu.kanade.tachiyomi.extension.all.namicomi.dto
+
+import eu.kanade.tachiyomi.extension.all.namicomi.NamiComiConstants
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.jsonPrimitive
+import kotlinx.serialization.serializer
+
+typealias MangaListDto = PaginatedResponseDto<MangaDataDto>
+
+typealias MangaDto = ResponseDto<MangaDataDto>
+
+@Serializable
+@SerialName(NamiComiConstants.manga)
+data class MangaDataDto(override val attributes: MangaAttributesDto? = null) : EntityDto()
+
+@Serializable
+data class MangaAttributesDto(
+    val title: LocalizedString,
+    val description: LocalizedString,
+    val slug: String,
+    val originalLanguage: String?,
+    val year: Int?,
+    val contentRating: ContentRatingDto? = null,
+    val publicationStatus: StatusDto? = null,
+    // val tags: List<TagDto>,
+) : AttributesDto()
+
+@Serializable
+enum class ContentRatingDto(val value: String) {
+    @SerialName("safe")
+    SAFE("safe"),
+
+    @SerialName("restricted")
+    RESTRICTED("restricted"),
+
+    @SerialName("mature")
+    MATURE("mature"),
+}
+
+@Serializable
+enum class StatusDto(val value: String) {
+    @SerialName("ongoing")
+    ONGOING("ongoing"),
+
+    @SerialName("completed")
+    COMPLETED("completed"),
+
+    @SerialName("hiatus")
+    HIATUS("hiatus"),
+
+    @SerialName("cancelled")
+    CANCELLED("cancelled"),
+}
+
+@Serializable
+abstract class AbstractTagDto(override val attributes: TagAttributesDto? = null) : EntityDto()
```

Comment:
```suggestion
sealed class AbstractTagDto(override val attributes: TagAttributesDto? = null) : EntityDto()
```
Similar case as above